Automattic/jetpack
# Instant Search: don't include highlight markup in alt text

Fixes #20188.

#### Changes proposed in this Pull Request:
Ensure HTML is stripped from alt text where images are displayed in Instant Search results.

Currently we generally use the post title as the alt text (until we can get the correct alt text into the index - see https://github.com/Automattic/jetpack/issues/19273). If the title matches a search keyword, it will contain a `<mark>` element. We do not want this element to appear in the image alt.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
1. On a site with Instant Search installed, select the 'Expanded' result format in the Customizer.
2. Add a test post with a featured image.
3. Search for the post by title - e.g. if the post is called "Hello World", search for "hello".
4. Inspect the image in the search result and ensure there are no `<mark>` tags in the output.
5. Repeat the same check with the 'Product' result format. You can use `&result_format=product` to check this.

<img width="1507" alt="Screen Shot 2021-06-28 at 12 44 16" src="https://user-images.githubusercontent.com/17325/123564728-9708a500-d80e-11eb-8d2e-7cfb65e09b9a.png">

Before:
<img width="234" alt="Screen Shot 2021-06-28 at 12 43 49" src="https://user-images.githubusercontent.com/17325/123564712-83f5d500-d80e-11eb-9d9e-6e71691cf466.png">

After:

<img width="155" alt="Screen Shot 2021-06-28 at 12 43 55" src="https://user-images.githubusercontent.com/17325/123564715-85bf9880-d80e-11eb-9f59-f0ea7001ace3.png">
